### ISSUE TITLE
influxdb.cardinality() returns sum of tag cardinalities instead of product of those

### ISSUE THREAD

**Issue by harcipulyka** [influxdata/flux#4755]
I am currently optimizing my database scheme, and wanted to check out the series cardinality of my bucket. Using the influxdb.cardinality() I get 91, but when I tried to double-check that number I found out this bug. In docs it say that this function should return the series cardinality of the bucket, which would be the possible number of unique combinations of tags in a bucket. Instead this function adds the cardinalities of each tag together.

Using the cardinalityByTag function that is in the documentation, I got the following result
<img width="1204" alt="Screenshot 2022-05-13 at 23 15 04" src="https://user-images.githubusercontent.com/55332570/168390506-c1a89580-18c5-4ebd-84cc-65a9c4fe3b5a.png">
You can see that we can only get 91 by adding the cardinalities of the tags instead of multiplying them. The actual one should be 88*4.

**Comment by sanderson** [comment]
@harcipulyka Series cardinality the number of **possible** combinations, it's the number of unique measurment, tag, and field key combinations stored on disk. `influxdb.cardility()` returns the cardinality of data in your bucket as it exists on disk.

For example, lets say you have the following data (represented as line protocol):

```
m,tag1=value1.1,tag2=value2.1 example-field1=2.1
m,tag1=value1.2,tag2=value2.2 example-field2=2.2
m,tag1=value1.3,tag3=value2.3 example-field3=2.3
```

If series cardinality represented the **possible** combinations, the cardinality of this data set would be 9:

*tag1:* `value1.1` | `value1.2` | `value1.3`
*tag2:* `value2.1` | `value2.2` | `value2.3`
*field:* `example-field1` | `example-field2` | `example-field3`

However, not all of these possible combinations actually exists on disk. The cardinality that exists on disk is 3.

By looking at tag cardinality, it shows you unique values for each tag, but you don't know how many unique combinations each tag exists in.

In short, `influxdb.cardinality()` is working as designed.
